packages/hoppscotch-common/src/components/smart/EnvInput.vue

```diff
@@ -357,25 +360,63 @@ const aggregateEnvs = useReadonlyStream(aggregateEnvsWithSecrets$, []) as Ref<
   AggregateEnvironment[]
 >
 
+const tabs = useService(RESTTabService)
+
 const envVars = computed(() => {
   return props.envs
     ? props.envs.map((x) => {
         if (x.secret) {
           return {
             key: x.key,
-            sourceEnv: "source" in x ? x.source : null,
+            sourceEnv: "sourceEnv" in x ? x.sourceEnv : null,
             value: "********",
+            secret: true,
           }
         }
         return {
           key: x.key,
           value: x.value,
-          sourceEnv: "source" in x ? x.source : null,
+          sourceEnv: "sourceEnv" in x ? x.sourceEnv : null,
+          secret: false,
         }
       })
-    : aggregateEnvs.value
+    : [
+        ...tabs.currentActiveTab.value.document.request.requestVariables.map(
+          (x) =>
+            x.active
+              ? {
+                  key: x.key,
+                  value: x.value,
+                  sourceEnv: "RequestVariable",
+                  secret: false,
+                }
+              : ([] as unknown as AggregateEnvironment)
+        ),
+        ...aggregateEnvs.value,
+      ]
 })
```

Comment:
```suggestion
const envVars = computed(() => {
  if (props.envs) {
    return props.envs.map((x) => {
      const { key, secret } = x

      const value = secret ? "********" : x.value
      const sourceEnv = "sourceEnv" in x ? x.sourceEnv : null

      return {
        key,
        value,
        sourceEnv,
        secret,
      }
    })
  }

  return [
    ...tabs.currentActiveTab.value.document.request.requestVariables.map(
      ({ active, key, value }) =>
        active
          ? {
              key,
              value,
              sourceEnv: "RequestVariable",
              secret: false,
            }
          : ({} as AggregateEnvironment)
    ),
    ...aggregateEnvs.value,
  ]
})
```

Can you check if the above works? I've converted the ternary-based approach to follow an early return-based one as I felt it enhances readability. Also, I've grouped the checks in the first block (`props.envs` is available).